app/controllers/concerns/params_store_concern.rb

```diff
@@ -0,0 +1,15 @@
+module ParamsStoreConcern
+  extend ActiveSupport::Concern
+
+  def store_params
+    return if session[:stored_params].present?
+
+    session[:stored_params] = params.to_unsafe_h.except(:controller, :action).to_h.to_yaml
+  end
+
+  def stored_params
+    return {} if session[:stored_params].blank?
+
+    YAML.load(session.delete(:stored_params))
```

Comment:
Voir https://rubyapi.org/3.1/o/psych#method-c-load

```suggestion
    YAML.safe_load(session.delete(:stored_params))
```

Reply:
On est d'accord qu'il n'y a pas d'équivalent avec JSON, et donc qu'il n'y a pas de risque à partir du moment où on sérialise avec JSON plutôt que YAML ?

Reply:
oui